packages/ui-react-utils/src/DeterministicIdContext/withDeterministicId.tsx

```diff
@@ -0,0 +1,101 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 - present Instructure, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+import React, {
+  forwardRef,
+  PropsWithoutRef,
+  RefAttributes,
+  useContext
+} from 'react'
+import hoistNonReactStatics from 'hoist-non-react-statics'
+
+import { DeterministicIdContext } from './DeterministicIdContext'
+import { decorator } from '@instructure/ui-decorator'
+import { generateId } from '@instructure/ui-utils'
+
+import type { InstUIComponent } from '@instructure/shared-types'
+import { warn } from '@instructure/console'
+
+type WithDeterministicIdProps = {
+  deterministicId?: (instanceName?: string) => string
+}
+/**
+ * This decorator is used to enable the decorated class to use the `DeterministicIdContext` which is needed
+ * for deterministic id generation.
+ *
+ * The context is there for the users to pass an `instanceCounterMap` Map which is then used
+ * in the child components to deterministically create ids for them based on the `instanceCounterMap`.
+ * Read more about it here: [SSR guide](/#server-side-rendering)
```

Comment:
I'd put here the whole link, otherwise it will break when opened via an IDE popup

Reply:
Sure, added the whole link